pipeline/trusted-resources/pkg/trustedtask/trustedtaskrun.go

```diff
@@ -101,125 +93,58 @@ func (tr *TrustedTaskRun) VerifyTask(
 		return apis.ErrMissingField(fmt.Sprintf("annotations[%s]", SignatureAnnotation))
 	}
 
-	signature, err := base64.StdEncoding.DecodeString(tr.ObjectMeta.Annotations[SignatureAnnotation])
+	cp, signature, err := copyTaskRun(tr)
 	if err != nil {
 		return apis.ErrGeneric(err.Error(), "metadata")
 	}
 
-	verifier, err := verifier(ctx, tr.ObjectMeta.Annotations, k8sclient)
+	verifier, err := verifier(ctx, cp.ObjectMeta.Annotations, k8sclient)
 	if err != nil {
 		return apis.ErrGeneric(err.Error(), "metadata")
 	}
 
-	if tr.Spec.TaskSpec != nil {
-		logger.Info("Verifying TaskSpec")
-		if err := VerifyInterface(ctx, tr.Spec.TaskSpec, verifier, signature); err != nil {
-			return apis.ErrGeneric(err.Error(), "spec")
-		}
-		return nil
+	if err := VerifyInterface(ctx, cp, verifier, signature); err != nil {
+		return apis.ErrGeneric(err.Error(), "taskrun")
 	}
 
 	if tr.Spec.TaskRef != nil {
-		if tr.Spec.TaskRef.Bundle != "" {
-			logger.Info("Verifying OCI Bundle")
-			if err := VerifyTaskOCIBundle(ctx, tr.Spec.TaskRef.Bundle, verifier, signature, k8sclient); err != nil {
-				return apis.ErrGeneric(err.Error(), "spec", "taskRef")
-			}
-			return nil
-		}
-
-		ts, err := tektonClient.TektonV1beta1().Tasks(tr.Namespace).Get(ctx, tr.Spec.TaskRef.Name, metav1.GetOptions{})
-		if err != nil {
-			return apis.ErrGeneric(err.Error(), "spec", "taskRef")
-		}
-		if ts.Name != "" {
-			logger.Info("Verifying TaskRef")
-			if err := VerifyInterface(ctx, &ts.Spec, verifier, signature); err != nil {
-				if err != nil {
-					return apis.ErrGeneric(err.Error(), "spec", "taskRef")
-				}
-			}
-			return nil
-		}
+		// TODO: add task verification in next PR
 	}
 
 	return nil
 }
 
-func verifier(
-	ctx context.Context,
-	annotations map[string]string,
-	k8sclient kubernetes.Interface,
-) (signature.Verifier, error) {
-	if annotations[kmsAnnotation] != "" {
-		// Fetch key from kms.
-		return kms.Get(ctx, annotations[kmsAnnotation], crypto.SHA256)
-	} else {
-		cosignPublicKeypath := secretPath
-		// Overwrite the path if set in configmap.
-		cm, err := k8sclient.CoreV1().ConfigMaps(system.Namespace()).Get(ctx, signingConfigMap, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
-		}
-		if cm.Data["signing-secret-path"] != "" {
-			cosignPublicKeypath = cm.Data["signing-secret-path"]
-		}
-		return cosignsignature.LoadPublicKey(ctx, cosignPublicKeypath)
-	}
-}
 
-func VerifyInterface(
-	ctx context.Context,
-	taskspec interface{},
-	verifier signature.Verifier,
-	signature []byte,
-) (errs *apis.FieldError) {
-	ts, err := json.Marshal(taskspec)
-	if err != nil {
-		return apis.ErrGeneric(err.Error(), "taskSpec")
-	}
+func copyTaskRun(in *TrustedTaskRun) (v1beta1.TaskRun, []byte, error) {
```

Comment:
We probably want to rename this (+ add func comments) this is really doing multiple things: 1) copying certain fields 2) removing certain data 3) extracting the signature.

1+2 probably makes sense together, 3 you may want to split out.

Reply:
yes thanks!